packages/lib/src/components/Badge/TaskStatusBadge/index.tsx

```diff
@@ -35,24 +35,18 @@ const TaskStatusBadge: React.FC<TaskStatusBadgeProps> = ({ status }) => {
 
   const statusText = taskStatusToCopy[status] ?? status;
 
-  if (status in mapTaskStatusToBadgeVariant) {
-    return (
-      <StyledBadge
-        key={status}
-        css={badgeWidthMaxContent}
-        data-cy="task-status-badge"
-        variant={mapTaskStatusToBadgeVariant[status]}
-      >
-        {statusText}
-      </StyledBadge>
-    );
-  }
-
   return (
     <StyledBadge
       key={status}
+      css={badgeWidthMaxContent}
       data-cy="task-status-badge"
-      {...customBadgeColors(status)}
+      variant={
+        status in mapTaskStatusToBadgeVariant
+          ? mapTaskStatusToBadgeVariant[status]
+          : undefined
+      }
+      {...(!(status in mapTaskStatusToBadgeVariant) &&
+        customBadgeColors(status))}
     >
```

Comment:
could it be simplified to:
```suggestion
      variant={mapTaskStatusToBadgeVariant[status]}
      {...customBadgeColors(status)}
    >
```